code/API_definitions/qod-api.yaml

```diff
@@ -325,20 +328,54 @@ components:
         * `QOS_S` - Qualifier for the requested QoS profile _S_
         * `QOS_M` - Qualifier for the requested QoS profile _M_
         * `QOS_L` - Qualifier for the requested QoS profile _L_
+    
     Notification:
       type: object
       required:
         - sessionId
-        - event
+        - eventReports
       properties:
         sessionId:
           $ref: "#/components/schemas/SessionId"
+        eventReports:
+          type: array
+          items:
+            $ref: "#/components/schemas/EventReport"
+          minItems: 1
+          description: Contains the reported event and applicable information
+    
+    EventReport:
+      type: object
+      required:
+        - event
+      properties:
         event:
           $ref: "#/components/schemas/SessionEvent"
+      discriminator:
+        propertyName: event
+        mapping:
+          QOS_STATUS_CHANGED: QosStatusChangedEventReport
+        
+    QosStatusChangedEventReport:
+      allOf:
+        - $ref: '#/components/schemas/EventReport'
+        - type: object
+          required:
+            - qosStatus
+          properties:
+            qoStatus:
+              $ref: "#/components/schemas/QosStatus"
+            eventInfo:
+              type: string
+              enum:
+                - DURATION_EXPIRED
+                - NETWORK_TERMINATED
+
     SessionEvent:
       type: string
       enum:
-        - SESSION_TERMINATED
```

Comment:
Is there no case for SESSION_TERMINATED anymore? If we say that the session resource is independent from the qosStatus of the session, if may be more accurate to tell that a session is terminated due to the duration expired, than  that the QoS changed to UNAVAILABLE due to it (as you say below   * `UNAVAILABLE` - The requested QoS cannot be provided by the network due to some reason).
This is subtle anyway, but as said before we should double check that all of this is understood from a developer perspective.